siac/cmds.go

```diff
@@ -47,20 +48,64 @@ func updatecmd(cmd *cobra.Command, args []string) {
 		cmd.Usage()
 		return
 	}
-	fmt.Println(getResponse("/update", &url.Values{
-		"action": {args[0]},
-	}))
+	switch args[0] {
+	case "check":
+		if m, err := getResponse("/update/check", nil); err == nil {
+			if m["Available"].(bool) {
+				fmt.Printf("Update %s is available!\nRun 'siac update %s' to install it.\n", m["Version"], m["Version"])
+			} else {
+				fmt.Println("Up to date!")
+			}
+		} else {
+			fmt.Println("Update check failed:", err)
+		}
+
+	default:
+		_, err := getResponse("/update/apply", &url.Values{
+			"version": {args[0]},
+		})
+		if err == nil {
```

Comment:
similar deal, I think we should standardize around always writing `if err != nil {`, because occasionally you get confused (like when I switched up the responses in a recent PR).

It makes the developer less likely to make errors, and it makes the reviewer less likely to miss errors.

We were arguing a bit about trusting developers, but humans are really, really error prone machines. And with code, a single `>` vs. `>=` or `!=` vs. `==` in the wrong place can be as severe as giving root access to any application that runs certain code (I believe this happened to the linux kernel at one point). Managing code with the expectation that the developer is going to make mistakes in random places is good practice, because it's the reality of programming.